swarm/network/stream/common_test.go

```diff
@@ -90,19 +96,24 @@ func NewStreamerService(ctx *adapters.ServiceContext) (node.Service, error) {
 	delivery := NewDelivery(kad, db)
 	deliveries[id] = delivery
 	r := NewRegistry(addr, delivery, db, state.NewMemStore(), &RegistryOptions{
-		SkipCheck: defaultSkipCheck,
+		SkipCheck:  defaultSkipCheck,
+		DoRetrieve: defaultDoRetrieve,
 	})
 	RegisterSwarmSyncerServer(r, db)
 	RegisterSwarmSyncerClient(r, db)
 	go func() {
 		waitPeerErrC <- waitForPeers(r, 1*time.Second, peerCount(id))
 	}()
-	dpa := storage.NewDPA(storage.NewNetStore(store, nil), storage.NewDPAParams())
+	dpa := storage.NewDPA(storage.NewNetStore(store, getRetrieveFunc(id)), storage.NewDPAParams())
 	testRegistry := &TestRegistry{Registry: r, dpa: dpa}
 	registries[id] = testRegistry
 	return testRegistry, nil
 }
 
+func defaultRetrieveFunc(id discover.NodeID) func(chunk *storage.Chunk) error {
```

Comment:
why is this needed?

Reply:
Because the other tests prior to this retrieval test had the retrieval func set to `nil`, while we need to have the `retrieveFunc` set for the retrieval tests